share/completions/terraform.fish

```diff
@@ -15,60 +15,193 @@ function __fish_terraform_needs_command
     return 0
 end
 
+function __fish_terraform_using_command
+    set cmd (commandline -opc)
+    if [ (count $cmd) -eq 1 ]
+        return 0
+    end
+
+    for arg in $cmd[2..-1]
+        switch $arg
+            case '--*'
+                # ignore global flags
+            case $argv[1]
+                return 0
+            case '*'
+                return 1
+        end
+    end
+    return 1
+end
+
 # general options
-complete -f -c terraform -n '__fish_terraform_needs_command' -l version -d 'Print version information'
-complete -f -c terraform -n '__fish_terraform_needs_command' -l help -d 'Show help'
+complete -f -c terraform -l version -d 'Print version information'
+complete -f -c terraform -l help -d 'Show help'
 
 ### apply
 complete -f -c terraform -n '__fish_terraform_needs_command' -a apply -d 'Builds or changes infrastructure'
+complete -f -c terraform -n '__fish_terraform_using_command apply' -o backup -d 'Path to backup the existing state file'
```

Comment:
Similar to my previous comment I believe the current best practice is to use `__fish_seen_subcommand_from` rather than define a custom function which does essentially the same thing.